codersdk/deployment.go

```diff
@@ -1348,12 +1348,18 @@ Write out the current server configuration to the path specified by --config.`,
 		{
 			// Env handling is done in cli.ReadGitAuthFromEnvironment
 			Name:        "Git Auth Providers",
-			Description: "Git Authentication providers",
+			Description: "Git Authentication providers.",
 			YAML:        "gitAuthProviders",
 			Value:       &c.GitAuthProviders,
 			Hidden:      true,
 		},
 	}
+	for i := range opts {
+		if opts[i].Env != "" {
+			opts[i].Env = "CODER_" + opts[i].Env
+		}
+	}
```

Comment:
This doesn't seem to be applied to all embedded properties, e.g. `UseInstead`. Perhaps we should just "type it out" like we do with flags? (Allows for use-cases where we have envs that don't start with CODER_ too.)

<img width="416" alt="image" src="https://user-images.githubusercontent.com/147409/225893623-cf9b30a0-00ed-4ed4-bbfe-dc121604d432.png">